tests/components/LocationBias.test.tsx

```diff
@@ -0,0 +1,148 @@
+import {fireEvent, render, screen, waitFor} from "@testing-library/react";
+import { LocationBias } from "../../src/components/LocationBias";
+import { State, LocationBiasMethod, useAnswersActions } from "@yext/answers-headless-react";
+import * as locationOperations from "../../src/utils/location-operations";
+import { mockAnswersHooks, mockAnswersState, spyOnActions } from "../__utils__/mocks";
+import * as searchOperations from "../../src/utils/search-operations";
+
+jest.mock("@yext/answers-headless-react");
+
+const mockedStateVaDevice: Partial<State> = {
+  location: {
+    locationBias: {
+      latitude: 38.89552025579547,
+      longitude: -77.06991520330985,
+      displayName: 'Arlington, VA', 
+      method: LocationBiasMethod.Device
+    }
+  }
+}
+
+const mockedStateNyIP: Partial<State> = {
+  location: {
+    locationBias: {
+      latitude: 40.741591687843005,
+      longitude: -74.00530254443494,
+      displayName: 'New York City, NY', 
+      method: LocationBiasMethod.Ip
+    }
+  }
+}
+
+const mockedStateNoDisplayName: Partial<State> = {
+  location: {
+    locationBias: {
+      latitude: null,
+      longitude: null,
+      displayName: null, 
+      method: null
+    }
+  },
+}
+
+const newGeoPosition = {
+  coords: {
+    accuracy: null,
+    altitude: null,
+    altitudeAccuracy: null,
+    heading: null,
+    latitude: 40.741591687843005,
+    longitude: -74.00530254443494,
+    speed: null,
+  },
+timestamp: null
+};
+
+describe("LocationBias", () => {
+  beforeEach(() => {
+    mockAnswersHooks({
+      mockedState: mockedStateVaDevice,
+      mockedActions: {
+        state: mockedStateVaDevice,
+        setUserLocation: jest.fn(),
+      } 
+    });
+    jest.spyOn(locationOperations, "getUserLocation").mockImplementation(async ()=> {
+      return Promise.resolve(newGeoPosition);
+    });
+    jest.spyOn(searchOperations, "executeSearch").mockImplementation();
+    // jest.spyOn(searchOperations, "executeSearch").mockImplementation(()=> new Promise(()=>null));
+  });
+
+  it("Proper text is rendered (Location name, method, and update btn)", () => {
+    render(<LocationBias />);
+    const expectedLocationName = mockedStateVaDevice.location.locationBias.displayName;
+    const locationNameElement = screen.getByText(expectedLocationName);
+    expect(locationNameElement).toBeDefined();
+
+    const expectedMethodMessage = "(based on your device) -";
+    const basedOnMethodElement = screen.getByText(expectedMethodMessage);
+    expect(basedOnMethodElement).toBeDefined();
+
+    const updateLocationButton = screen.getByText("Update your location");
+    expect(updateLocationButton).toBeDefined();
+  })
+
+  it("On update location click, setUserLocation is called with params being coordinates returned by getUserLocation", async () => {
```

Comment:
Wording a little weird here, specifically the "with params being".  Naming is hard.

We don't follow this rule a lot of the time, but one way to name your tests is as a readable sentence USING the "it" call as the first word. For example, if I had the test
`it('throws an error when I eat a cookie')`, you could read it as "It throws an error when I eat a cookie."
Again we probably follow this less than 50% of the time

Reply:
renamed